internal/resources/policy/kind/mutation/recipe/annotation_schema.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+// nolint: dupl
+package recipe
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	policyrecipecustomcommonmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/custom/common"
+	policyrecipemutationmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/mutation"
+	policyrecipemutationcommonmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/mutation/common"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy/kind/common"
+)
+
+var AnnotationSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "The input schema for custom policy tmc_block_nodeport_service recipe version v1",
+	Optional:    true,
+	ForceNew:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			targetKubernetesResourcesKey: common.TargetKubernetesResourcesSchema,
+			scopeKey: {
+				Type:         schema.TypeString,
+				Description:  "Audit (dry-run).",
+				Optional:     true,
+				Default:      "All",
+				ValidateFunc: validation.StringInSlice([]string{"All", "Cluster", "Namespaced"}, false),
+			},
+			AnnotationKey: {
+				Type:     schema.TypeList,
```

Comment:
As discussed this need not be a list of key-value pairs. Only one key-value pair can be provided using a map.

Reply:
Actually the TypeMap is not suitable for this. Because we need to support following:
```
    "annotation": {
        "key": "key_key",
        "value": "value_value"
    }
```
While TypeMap would be more suitable for following:
```
    "annotation": {
        "key": "key_key",
        "value": "value_value"
    }
```
 I tried to have TypeMap with Resource too, but this was not allowed. Got following error:
`* resource tanzu-mission-control_mutation_policy: label: TypeMap with Elem *Resource not supported,use TypeList/TypeSet with Elem *Resource or TypeMap with Elem *Schema`

So I would leave it as it is if there are no other options.